src/main/java/immibis/tubestuff/AutoCraftingMk2CraftingRecipeProvider.java

```diff
@@ -0,0 +1,77 @@
+package immibis.tubestuff;
+
+import logisticspipes.proxy.interfaces.ICraftingRecipeProvider;
+import logisticspipes.utils.ItemIdentifier;
+import logisticspipes.utils.SimpleInventory;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.tileentity.TileEntity;
+
+public class AutoCraftingMk2CraftingRecipeProvider implements ICraftingRecipeProvider {
```

Comment:
I know it's a big ask, but is TubeStuff the right mod to add this to? Logistics Pipes already includes a lot of cross-mod compatibility code, and UpsilonFixes seems to share the helper method `mergeStacks` across crafting recipe providers...

Reply:
This requires access to private members (`cachedRecipe`) which is the main reason why I commited this here. AT might work, though I never used them with Voldeloom.

`mergeStacks` was originally part of the XyCraft Fabricator recipe provider, but I moved it into a helper class when I made this, because both required that.